diff --git a/utils/SLHALib-2.2/makefile.in b/utils/SLHALib-2.2/makefile.in
index b9dede0..23391f2 100644
--- a/utils/SLHALib-2.2/makefile.in
+++ b/utils/SLHALib-2.2/makefile.in
@@ -15,9 +15,9 @@ all: default demo
 
 install: default
 	-mkdir $(PREFIX)
-	-mkdir $(PREFIX)/lib $(PREFIX)/include $(PREFIX)/bin
+	-mkdir -p $(PREFIX)/lib $(PREFIX)/include/SLHALib $(PREFIX)/bin
 	cp -p $(LIB) $(PREFIX)/lib
-	cp -p $(INCLUDE) $(PREFIX)/include
+	cp -p $(INCLUDE) $(PREFIX)/include/SLHALib
 	cp -p $(FCC) $(PREFIX)/bin
 	cd $(PREFIX)/bin ; ln -s fcc f++
 
diff --git a/utils/SLHALib-2.2/configure b/utils/SLHALib-2.2/configure
index 7058799..72b3853 100755
--- a/utils/SLHALib-2.2/configure
+++ b/utils/SLHALib-2.2/configure
@@ -259,7 +259,7 @@ DEF = $CONF_DEF
 UNDERSCORE = $CONF_UNDERSCORE
 
 FC = $CONF_FC
-FFLAGS = $CONF_FFLAGS
+FFLAGS = $CONF_FFLAGS -fPIC
 
 CC = $CONF_CC
 CFLAGS = $CONF_CFLAGS
